svc-api/handle/systems_test.go

```diff
@@ -469,3 +469,94 @@ func TestSetDefaultBootOrderWithValidData(t *testing.T) {
 		"/redfish/v1/Systems/123:1/Actions/ComputerSystem.SetDefaultBootOrder",
 	).WithJSON(map[string]string{"Sample": "Body"}).WithHeader("X-Auth-Token", "ValidToken").Expect().Status(http.StatusOK)
 }
+
+// Create volume unit tests
+func mockCreateVolume(req systemsproto.CreateVolumeRequest) (*systemsproto.SystemsResponse, error) {
+	var response = &systemsproto.SystemsResponse{}
+	if req.SessionToken == "" {
+		response = &systemsproto.SystemsResponse{
+			StatusCode:    401,
+			StatusMessage: "Unauthorized",
+			Body:          []byte(`{"Response":"Unauthorized"}`),
+		}
+	} else if req.SessionToken == "InvalidToken" {
+		response = &systemsproto.SystemsResponse{
+			StatusCode:    401,
+			StatusMessage: "Unauthorized",
+			Body:          []byte(`{"Response":"Unauthorized"}`),
+		}
+	} else if req.SessionToken == "ValidToken" && req.SystemID == "" {
+		response = &systemsproto.SystemsResponse{
+			StatusCode:    400,
+			StatusMessage: "BadRequest",
+			Body:          []byte(`{"Response":"BadRequest"}`),
+		}
+	} else if req.SessionToken == "TokenRPC" {
+		return &systemsproto.SystemsResponse{}, errors.New("Unable to RPC Call")
+	} else {
+		response = &systemsproto.SystemsResponse{
+			StatusCode:    200,
+			StatusMessage: "Success",
+			Body:          []byte(`{"Response":"Success"}`),
+		}
+	}
+	return response, nil
+}
+
+func TestCreateVolume(t *testing.T) {
+	var sys SystemRPCs
+	sys.CreateVolumeRPC = mockCreateVolume
+	mockApp := iris.New()
+	redfishRoutes := mockApp.Party("/redfish/v1/Systems")
+	redfishRoutes.Post("/{id}/Storage/{rid}/Volumes", sys.CreateVolume)
+
+	e := httptest.New(t, mockApp)
+	e.POST(
+		"/redfish/v1/Systems/6d4a0a66-7efa-578e-83cf-44dc68d2874e:1/Storage/ArrayControllers-0/Volumes",
+	).WithJSON(map[string]string{"Sample": "Body"}).WithHeader("X-Auth-Token", "ValidToken").Expect().Status(http.StatusOK)
```

Comment:
change expected status code StatusNotImplemented and  add to comments change the status code with actual status code and after code is added update the status code

Reply:
Addressed the comment